### PR TITLE
python3Packages.paver/nose: remove nose and modernize

### PR DESCRIPTION
**Pull request by lavafroth**
## Description of changes

Addresses #326513. Disables new tests that require distutils.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Sigmanificient — APPROVED** [review]

**Comment by Sigmanificient** [comment]
What about adding https://github.com/paver/paver/commit/e71a7355bbf7482eb5f785790c73c3c2bb218985 to remove `six` dependency?

**Comment by lavafroth** [comment]
Not sure if it's worth it. The commit version could be buggy for all we know.

File: pkgs/development/python-modules/paver/default.nix [hunk 1]
```diff
@@ -24,21 +24,24 @@ buildPythonPackage rec {
     hash = "sha256-0+ZJiIFIWrdQ7+QMUniYKpNDvGJ+E3sRrc7WJ3GTCMc=";
   };
 
+  build-system = [ setuptools ];
+
   propagatedBuildInputs = [ six ];
```
Comment (gador) [inline pkgs/development/python-modules/paver/default.nix:29]:
```suggestion
  dependencies = [ six ];
```

File: pkgs/development/python-modules/paver/default.nix [hunk 2]
```diff
@@ -24,21 +24,24 @@ buildPythonPackage rec {
...
 
   disabledTestPaths = [
-    # Test depends on distutils
+    # Tests depend on distutils
     "paver/tests/test_setuputils.py"
+    "paver/tests/test_doctools.py"
+    "paver/tests/test_tasks.py"
   ];
 
   meta = with lib; {
```
Comment (gador) [inline pkgs/development/python-modules/paver/default.nix:47]:
```suggestion
  meta = {
```

**Review by gador — CHANGES_REQUESTED** [review]
If we modernize `paver` we might as well change `with lib; ` and `dependencies`

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 330858` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python311Packages.paver</li>
<li>python311Packages.paver.dist</li>
<li>python312Packages.paver</li>
<li>python312Packages.paver.dist</li>
</ul>
</details>

**Review by gador — APPROVED** [review]

**Pull request merged**